packages/chat-component/src/main.ts

```diff
@@ -105,8 +108,29 @@ export class ChatComponent extends LitElement {
 
   static override styles = [mainStyle];
 
-  // Send the question to the Open AI API and render the answer in the chat
+  override connectedCallback(): void {
+    super.connectedCallback();
 
+    // add event listener to know when to scroll chat list footer into view
+    this.addEventListener('must-scroll', () => {
```

Comment:
Why use an event for that, and not perform the scroll directly in the debounced call?

Is this something that we want to expose to the outside?

Reply:
I didn't initially implement it as an event, however considering that the application is meant to be integrated as part of a shell application, I considered posting the event as a message to do things like closing menus or other type of UI improvements in the shell. We can handle that as a different issue and trigger and only once event when the item is into view already. Removed all the utils and event.